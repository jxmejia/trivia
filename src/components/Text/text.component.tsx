import { memo } from "react";
import type { TextProps } from "./text.definition";
import { StyledText } from "./text.style";

const Text = ({ children }: TextProps): JSX.Element => {
  return <StyledText>{children}</StyledText>;
};

export default memo(Text);
