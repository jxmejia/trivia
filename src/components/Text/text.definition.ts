import { ReactNode } from "react";

export type TextProps = {
  children?: ReactNode;
};
