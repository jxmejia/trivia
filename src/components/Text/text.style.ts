import styled from "@emotion/styled";

export const StyledText = styled.span`
  font-size: 14px;
`;
