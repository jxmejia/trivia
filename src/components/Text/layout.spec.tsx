import { createSerializer } from "@emotion/jest";
import { render } from "@testing-library/react";
import Text from "./text.component";

expect.addSnapshotSerializer(createSerializer());

describe("Text", () => {
  it("renders component", () => {
    const { container } = render(<Text />);
    expect(container).toMatchSnapshot();
  });
});
