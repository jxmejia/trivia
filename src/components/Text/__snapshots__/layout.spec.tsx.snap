// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text is renders component 1`] = `
.emotion-0 {
  font-size: 14px;
}

<div>
  <span
    class="emotion-0"
  />
</div>
`;

exports[`Text renders component 1`] = `
.emotion-0 {
  font-size: 14px;
}

<div>
  <span
    class="emotion-0"
  />
</div>
`;
